Add support for MySQL connections over UNIX sockets

dsbaars opened the pull request:
As described in #1359 this adds support for connecting to the database over UNIX sockets which is considered to be more secure.

Closes #1359

wiz commented:
Hey @dsbaars thanks for the PR. We were planning to add this so it's really great that you beat us to it :)

Would you mind rebasing onto current master to fix to merge conflicts and also squashing your PR branch a bit to cleanup those extra commits?

dsbaars commented:
> Hey @dsbaars thanks for the PR. We were planning to add this so it's really great that you beat us to it :)
> 
> Would you mind rebasing onto current master to fix to merge conflicts and also squashing your PR branch a bit to cleanup those extra commits?

Rebased and squashed!

softsimon reviewed (approved):
utACK
